src/test/javascript/spec/component/tutorial-groups/tutorial-groups-management.component.spec.ts

```diff
@@ -0,0 +1,114 @@
+                MockRouterLinkDirective,
+                MockDirective(SortDirective),
+                MockDirective(SortByDirective),
+            ],
+            providers: [
+                MockProvider(TutorialGroupsService),
+                MockProvider(AlertService),
+                MockProvider(SortService),
+                { provide: Router, useClass: MockRouter },
+                {
+                    provide: ActivatedRoute,
+                    useValue: {
+                        parent: {
+                            paramMap: of({
+                                get: (key: string) => {
+                                    switch (key) {
+                                        case 'courseId':
+                                            return 1;
+                                    }
+                                },
+                            }),
+                        },
+                    },
+                },
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                tutorialGroupsManagementComponentFixture = TestBed.createComponent(TutorialGroupsManagementComponent);
+                tutorialGroupsManagementComponent = tutorialGroupsManagementComponentFixture.componentInstance;
+            });
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    it('should initialize', () => {
+        tutorialGroupsManagementComponentFixture.detectChanges();
+        expect(tutorialGroupsManagementComponent).not.toBeNull();
+    });
+
+    it('should get all tutorial groups for course', () => {
+        const exampleTutorialGroup = new TutorialGroup();
+        exampleTutorialGroup.id = 1;
+
+        const tutorialGroupsService = TestBed.inject(TutorialGroupsService);
+        const getAllForCourseSpy = jest.spyOn(tutorialGroupsService, 'getAllOfCourse').mockReturnValue(
+            of(
+                new HttpResponse({
+                    body: [exampleTutorialGroup],
+                    status: 200,
+                }),
+            ),
+        );
+
+        tutorialGroupsManagementComponentFixture.detectChanges();
+        expect(tutorialGroupsManagementComponent.tutorialGroups).toEqual([exampleTutorialGroup]);
+        expect(getAllForCourseSpy).toHaveBeenCalledWith(1);
+    });
+
+    it('should call sort service', () => {
+        const group1 = new TutorialGroup();
+        group1.id = 1;
+        const group2 = new TutorialGroup();
+        group2.id = 2;
+
+        tutorialGroupsManagementComponent.tutorialGroups = [group1, group2];
+        tutorialGroupsManagementComponent.sortingPredicate = 'id';
+        tutorialGroupsManagementComponent.ascending = false;
+
+        const sortService = TestBed.inject(SortService);
+        const sortServiceSpy = jest.spyOn(sortService, 'sortByProperty');
+
+        tutorialGroupsManagementComponent.sortRows();
+        expect(sortServiceSpy).toHaveBeenCalledWith([group1, group2], 'id', false);
```

Comment:
Missing `toHaveBeenCalledOnce()`

Reply:
Done in 26527db8ac4e1b72878a831b3e1e749b67b59361